doc/development/html_themes/index.rst

```diff
@@ -221,6 +221,60 @@ If your theme package contains two or more themes, please call
    ``sphinx.html_themes`` entry_points feature.
 
 
+Styling with CSS
+----------------
+The **stylesheets** setting allows to add custom CSS files to a theme.
+
+The structure of the HTML elements and their classes are currently not a
+well-defined public API. Please infer them from inspecting the built HTML
+pages. While we cannot guarantee full stability, they tend to be fairly
+stable.
+
+Styling search result entries by category
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 8.0
+
+The search result items have classes indicating the context in which the
+search term was found. You can use the CSS selectors:
+
+- ``ul.search li.context-index``:
+  If the search term is found in an index such as the glossary
+- ``ul.search li.context-object``:
+  If the search term a code object such as a python function
+- ``ul.search li.context-text``:
+  If the search term is found in plain documentation text
+- ``ul.search li.context-title``:
+  If the search term is found in a heading
+
+The basic theme is intentional minimal and does not use them for styling.
```

Comment:
```suggestion
As a base for inheritance by other themes, the ``basic`` theme is intentionally minimal and does not define CSS rules using these.
```

(maybe too waffly, but I felt the additional background could be useful, despite the fact that we mention derived themes next-sentence)